PolyFill.cs

```diff
@@ -0,0 +1,60 @@
+﻿//// PolyFill.cs - Contains the PolyFill class which is mainly used to 
+//// fill a given polygon
+//// ---------------------------------------------------------------------------------------
+namespace GrayBMP;
+
+#region class PolyFill ---------------------------------------------------------
+/// <summary>Class to read a given polygon and to fill it </summary>
+class PolyFill {
+   public void AddLine (int x0, int y0, int x1, int y1) {
+      if (y0 == y1) return; // Don't add horizontal lines
+      // We will try to keep Y0 as the smaller one always
+      if (y0 > y1) { (y0, y1) = (y1, y0); (x0, x1) = (x1, x0); }
+      mLines.Enqueue (new Line (x0, y0, x1, y1));

```

Comment:
What's the point of a PriorityQueue for this? Why not just put the lines in a simple List and do a Sort? That would be just as efficient. If you are not going to interleave adding and removing of elements, a PriorityQueue has no advantage over a simple sorted list.

Reply:
@tarydon: Checked the performance between using a simple list and a priority queue. The time difference is very small; around 3 ms. I used the priority queue as to sort the elements on enqueuing them. Thanks for reviewing